src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -57,45 +54,31 @@ internal class RazorDefinitionEndpoint : IDefinitionEndpoint
             return new RegistrationExtensionResult(ServerCapability, option);
         }
 
-        public async Task<DefinitionResult?> Handle(DefinitionParamsBridge request, CancellationToken cancellationToken)
+        protected async override Task<DefinitionResult?> TryHandleAsync(DefinitionParamsBridge request, DocumentContext documentContext, Projection projection, CancellationToken cancellationToken)
         {
-            _logger.LogInformation("Starting go-to-def endpoint request.");
-
-            if (request is null)
-            {
-                _logger.LogWarning("Request is null.");
-                throw new ArgumentNullException(nameof(request));
-            }
-
-            var documentContext = await _documentContextFactory.TryCreateAsync(request.TextDocument.Uri, cancellationToken).ConfigureAwait(false);
-
-            if (documentContext is null)
-            {
-                _logger.LogWarning("Document snapshot is null for document.");
-                return null;
-            }
+            Logger.LogInformation("Starting go-to-def endpoint request.");
```

Comment:
Side thought: We should eventually nuke these types of logs and rely on the CLaSP equivalent logging to indicate what's being called